rust/private/utils.bzl

```diff
@@ -296,3 +296,30 @@ def dedent(doc_string):
         # Remove the leading block of spaces from the current line
         block = " " * space_count
         return "\n".join([line.replace(block, "", 1).rstrip() for line in lines])
+
+def make_static_lib(ctx, rlib_file):
+    """Add a .a symlink to an .rlib file, so it can be used as a staticlib.
+
+    The name of the symlink is derived from the <name> of the <name>.rlib file as follows:
+    * `<name>.a`, if <name> starts with `lib`
+    * `lib<name>.a`, otherwise.
+
+    For example, the name of the symlink for
+    * `libcratea.rlib` is `libcratea.a`
+    * `crateb.rlib` is `libcrateb.a`.
+
+    Args:
+        ctx (ctx): The rule's context object.
+        rlib_file (File): The file to symlink, which must end in .rlib.
+
+    Returns:
+        The symlink's File.
+    """
+    if not rlib_file.basename.endswith(".rlib"):
+        fail("file is not an .rlib: ", rlib_file.basename)
+    basename = rlib_file.basename[:-5]
+    if not basename.startswith("lib"):
+        basename = "lib" + basename
+    dot_a = ctx.actions.declare_file(basename + ".a", sibling = rlib_file)
+    ctx.actions.symlink(output = dot_a, target_file = rlib_file)
```

Comment:
SuperNit: this function actually only needs ctx.actions, not the whole ctx. Let's pass actions there.